op-supervisor/supervisor/backend/source/chain.go

```diff
@@ -43,26 +45,26 @@ func NewChainMonitor(ctx context.Context, logger log.Logger, m Metrics, chainID
 		return nil, err
 	}
 
-	latest, ok := store.LatestBlockNum(chainID)
-	if !ok {
-		logger.Warn("")
-	}
+	// Create the log processor and fetcher
+	processLogs := newLogProcessor(chainID, store)
+	unsafeBlockProcessor := NewChainProcessor(logger, cl, chainID, processLogs, store)
 
-	startingHead := eth.L1BlockRef{
-		Number: latest,
-	}
+	// create head processors which only update the head
+	unsafeHeadProcessor := OnNewHead(chainID, store.Heads().UpdateLocalUnsafe)
+	safeHeadProcessor := OnNewHead(chainID, store.Heads().UpdateLocalSafe)
+	finalizedHeadProcessor := OnNewHead(chainID, store.Heads().UpdateLocalFinalized)
 
-	processLogs := newLogProcessor(chainID, store)
-	fetchReceipts := newLogFetcher(cl, processLogs)
-	unsafeBlockProcessor := NewChainProcessor(logger, cl, chainID, startingHead, fetchReceipts, store)
+	unsafeProcessors := []HeadProcessor{unsafeBlockProcessor, unsafeHeadProcessor}
+	safeProcessors := []HeadProcessor{safeHeadProcessor}
+	finalizedProcessors := []HeadProcessor{finalizedHeadProcessor}
 
-	unsafeProcessors := []HeadProcessor{unsafeBlockProcessor}
-	callback := newHeadUpdateProcessor(logger, unsafeProcessors, nil, nil)
+	callback := newHeadUpdateProcessor(logger, unsafeProcessors, safeProcessors, finalizedProcessors)
 	headMonitor := NewHeadMonitor(logger, epochPollInterval, cl, callback)
 
 	return &ChainMonitor{
-		log:         logger,
-		headMonitor: headMonitor,
+		log:            logger,
+		headMonitor:    headMonitor,
+		chainProcessor: unsafeBlockProcessor,
```

Comment:
Not sure if this is intentional, but the `unsafeBlockProcessor` is now embedded twice:
* Once as this new `chainProcessor`
* Once inside `headMonitor` <- `callback` <- `unsafeProcessors`

Looks like this new `chainProcessor` value is just so we call `Close` on it? There's already `headMonitor.Stop` so I'm not sure the purpose of this.